kv/leader_cache.go

```diff
@@ -63,7 +63,10 @@ func (lc *leaderCache) Update(group proto.RaftID, r *proto.Replica) {
 	lc.mu.Lock()
 	lc.cache.Del(group)
 	if r != nil {
-		lc.cache.Add(group, r)
+		// Replicas reside inside of slices that are subject to rearrangements,
```

Comment:
Maybe this method should just take a Replica instead of a `*Replica` so the copying is automatic. The structs are small; it shouldn't be expensive to just pass them on the stack.